### PR TITLE
Use parameterization in test_rank_adjustment

### PR DESCRIPTION
**Pull request by andersendsa**
## Description

Refactored test_rank_adjustment in keras/src/losses/loss_test.py to use pytest.mark.parametrize instead of a manual for loop. Removed associated TODO and commented-out code.

## Contributor Agreement

**Please review our [AI-Assisted Contribution Policy](../CONTRIBUTING.md#ai-assisted-contribution-policy) and check all boxes below before submitting your PR for review:**

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

**Note:** Failing to adhere to this agreement may result in your future PRs no longer being reviewed.

### CODE REVIEW

File: keras/src/losses/loss_test.py [hunk 1]
```diff
@@ -184,47 +184,42 @@ def test_mask_and_sample_weight_rank2(self):
         rank2_loss = loss_fn(y_true, y_pred, sample_weight=sample_weight)
         self.assertAllClose(rank1_loss, rank2_loss)
 
-    # @testing.parametrize(
-    #     "uprank", ["mask", "sample_weight", "y_true", "y_pred"])
-    # TODO: use parameterization decorator
     @pytest.mark.skipif(
         backend.backend() == "numpy",
         reason="Numpy backend does not support masking.",
     )
```
Comment (hertschuh) [inline keras/src/losses/loss_test.py]: Remove this, masking support was added to Numpy.

File: keras/src/losses/loss_test.py [hunk 2]
```diff
@@ -184,47 +184,42 @@ def test_mask_and_sample_weight_rank2(self):
...
-            self.assertEqual(backend.standardize_dtype(loss.dtype), "float32")
-            self.assertAllClose(
-                np.sum(
-                    masked_sample_weight * (masked_y_true - masked_y_pred) ** 2
-                )
-                / 3,
-                loss,
-            )
+    @pytest.mark.parametrize("uprank", ["mask", "sample_weight", "ys"])
+    def test_rank_adjustment(self, uprank):
```
Comment (hertschuh) [inline keras/src/losses/loss_test.py]: You have the wrong syntax here, you need:
```python
from absl.testing import parameterized
...

    @parameterized.named_parameters(
        ("mask", "mask"),
        ("sample_weight", "sample_weight"),
        ("ys", "ys"),
    )
    def test_rank_adjustment(self, uprank):
        ...
```

**Review by hertschuh — CHANGES_REQUESTED** [review]

File: keras/src/losses/loss_test.py [hunk 3]
```diff
@@ -184,47 +185,44 @@ def test_mask_and_sample_weight_rank2(self):
...
-            self.assertAllClose(
-                np.sum(
-                    masked_sample_weight * (masked_y_true - masked_y_pred) ** 2
-                )
-                / 3,
-                loss,
-            )
+    def test_rank_adjustment(self, uprank):
+        if backend.backend() == "numpy":
+            self.skipTest("Numpy backend does not support masking.")
```
Comment (hertschuh) [inline keras/src/losses/loss_test.py]: Remove, this should work as numpy supports masking.

**Review by hertschuh — CHANGES_REQUESTED** [review]

**Comment by andersendsa** [comment]
@hertschuh i have done the changes the pr is ready for review

**Review by hertschuh — APPROVED** [review]
Thanks!
